libraries/botbuilder-schema/botbuilder/schema/teams/_models_py3.py

```diff
@@ -2058,3 +2086,265 @@ def __init__(
+        self.cards = cards
+        self._custom_init()
+
+    def _custom_init(self):
+        return
+
+
+class TabResponsePayload(Model):
+    """
+    Initializes a new instance of the TabResponsePayload class.
+
+    :param type: Gets or sets choice of action options when responding to the
+     tab/fetch message. Possible values include: 'continue', 'auth' or 'silentAuth'
+    :type type: str
+    :param value: Gets or sets the TabResponseCards when responding to 
+     tab/fetch activity with type of 'continue'.
+    :type value: TabResponseCards
+    :param suggested_actions: Gets or sets the Suggested Actions for this card tab.
+    :type suggested_actions: TabSuggestedActions
+    """
+
+    _attribute_map = {
+        "type": {"key": "type", "type": "str"},
+        "value": {"key": "value", "type": "TabResponseCards"},
+        "suggested_actions": {"key": "suggestedActions", "type": "TabSuggestedActions"},
+    }
+
+    def __init__(
+        self, *, type=None, value=None, suggested_actions=None, **kwargs
+    ) -> None:
+        super(TabResponsePayload, self).__init__(**kwargs)
+        self.type = type
+        self.value = value
+        self.suggested_actions = suggested_actions
+        self._custom_init()
+
+    def _custom_init(self):
+        return
+
+
+class TabSumit(Model):
+    """
+    Invoke ('tab/submit') request value payload.
+
+    :param tab_entity_context: Gets or sets current tab entity request context.
+    :type tab_entity_context:
+     ~botframework.connector.teams.models.TabEntityContext
+    :param context: Gets or sets current tab entity request context.
+    :type context:
+     ~botframework.connector.teams.models.TabContext
+    :param data: User input data. Free payload containing properties of key-value pairs.
+    :type data:
+     ~botframework.connector.teams.models.TabSubmitData
+    """
+
+    _attribute_map = {
+        "tab_entity_context": {"key": "tabEntityContext", "type": "TabEntityContext"},
+        "context": {"key": "context", "type": "TabContext"},
+        "data": {"key": "data", "type": "TabSubmitData"},
+    }
+
+    def __init__(
+        self, *, tab_entity_context=None, context=None, data=None, **kwargs
+    ) -> None:
+        super(TabSumit, self).__init__(**kwargs)
+        self.tab_entity_context = tab_entity_context
+        self.context = context
+        self.data = data
+        self._custom_init()
+
+    def _custom_init(self):
+        return
+
+
+class TabSubmitData(Model):
+    """
+    Invoke ('tab/submit') request value payload data.
+
+    :param type: Currently, 'tab/submit'.
+    :type type: str
+    :param properties: Gets or sets properties that are not otherwise defined by the TabSubmit
+     type but that might appear in the serialized REST JSON object.
+    :type properties: object
+    """
+
+    _attribute_map = {
+        "type": {"key": "type", "type": "str"},
+        "properties": {"key": "properties", "type": "object"},
```

Comment:
I don't believe this will parse correctly. The [spec](https://github.com/microsoft/botframework-sdk/issues/6138) is:

```json
{
    "name": "tab/submit",
    "value": {
        "data": {
            "type": "tab/submit",
            // ... <data properties>
        },
        [...]
}
```

My understanding is that there won't be a `properties` key, so much as an unknown amount of un-named keys. @EricDahlvang @clearab Is this correct?

If so, [here's how I handled it in JS](https://github.com/microsoft/botbuilder-js/pull/3224/files#diff-348dcf11b9d343d90e766394f8ebae3c3675bde51e95705520bd15451f0c5720R1365) (not sure what the Py equivalent is).

Reply:
This should be handled similarly to the Properties field on Entity, ConversationAccount, ChannelAccount, Activity, etc.